azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecutorServer.java

```diff
@@ -134,34 +171,35 @@ public AzkabanExecutorServer(Props props) throws Exception {
     root.addServlet(new ServletHolder(new StatsServlet()), "/stats");
     root.addServlet(new ServletHolder(new ServerStatisticsServlet()), "/serverStatistics");
 
-    root.setAttribute(ServerConstants.AZKABAN_SERVLET_CONTEXT_KEY, this);
-
-    executionLoader = createExecLoader(props);
-    projectLoader = createProjectLoader(props);
-    runnerManager =
-        new FlowRunnerManager(props, executionLoader, projectLoader, this
-            .getClass().getClassLoader());
-
-    JmxJobMBeanManager.getInstance().initialize(props);
-
-    // make sure this happens before
-    configureJobCallback(props);
-
-    configureMBeanServer();
-    configureMetricReports();
-
-    SystemMemoryInfo.init(props.getInt("executor.memCheck.interval", 30));
-
-    loadCustomJMXAttributeProcessor(props);
+    root.setAttribute(Constants.AZKABAN_SERVLET_CONTEXT_KEY, this);
+    return server;
+  }
 
+  private void insertExecutorEntryIntoDB() {
     try {
-      server.start();
-    } catch (Exception e) {
-      logger.warn(e);
-      Utils.croak(e.getMessage(), 1);
+      final String host = requireNonNull(getHost());
+      final int port = getPort();
+      checkState(port != -1);
+      final Executor executor = executionLoader.fetchExecutor(host, port);
+      if (executor == null) {
+        executionLoader.addExecutor(host, port);
+      }
+      // If executor already exists, ignore it
+    } catch (ExecutorManagerException e) {
+      logger.error("Error inserting executor entry into DB", e);
+      Throwables.propagate(e);
     }
+  }
 
-    logger.info("Azkaban Executor Server started on port " + portNumber);
+  private void dumpPortToFile() {
+    // By default this should write to the working directory
+    String filename = props.getString(AZKABAN_EXECUTOR_PORT_FILE, "executor.port");
```

Comment:
No. In fact I don't see any advantage in doing it.
I just did it since I didn't see any disadvantage either. It allows people to put whatever name they like.
This can be changed to a constant.